airflow/metrics/validators.py

```diff
@@ -85,6 +85,42 @@ class MetricNameLengthExemptionWarning(Warning):
 BACK_COMPAT_METRIC_NAMES: set[Pattern[str]] = {re2.compile(name) for name in BACK_COMPAT_METRIC_NAME_PATTERNS}
 
 OTEL_NAME_MAX_LENGTH = 63
+DEFAULT_VALIDATOR_TYPE = "allow"
+
+
+def get_validator() -> ListValidator:
+    validators = {
+        "basic": {"allow": AllowListValidator, "block": BlockListValidator},
+        "pattern": {"allow": PatternAllowListValidator, "block": PatternBlockListValidator},
+    }
+    metric_lists = {
+        "allow": (metric_allow_list := conf.get("metrics", "metrics_allow_list", fallback=None)),
+        "block": (metric_block_list := conf.get("metrics", "metrics_block_list", fallback=None)),
+    }
+
+    use_pattern = conf.getboolean("metrics", "metrics_use_pattern_match", fallback=False)
+    validator_type = "pattern" if use_pattern else "basic"
+
+    if not use_pattern:
+        warnings.warn(
+            "The basic metric validator will be deprecated in the future in favor of pattern-matching.  "
+            "You can try this now by setting config option metrics_use_pattern_match to True.",
+            AirflowProviderDeprecationWarning,
```

Comment:
This is a part of Airflow core, you cannot use `AirflowProviderDeprecationWarning` warning:
```suggestion
            RemovedInAirflow3Warning,
```